tests/methods/test_lora.py

```diff
@@ -14,9 +19,262 @@ def test_leave_out_lora(self):
+        # collect weights of the first adapter so we can compare them to the newly created adapters in the subsequent tests
+        filter_keys_adapter_0 = [k.format(name=f"{name}_0") for k in ["loras.{name}."]]
+        adapter_0 = self.filter_parameters(model, filter_keys_adapter_0)
+
+        # Run tests for every combine strategy
+        for combine_strategy in ["linear", "lora_linear_only_negate_b", "lora_delta_w_svd"]:
+            if not model_supports_lora_delta_w_svd and combine_strategy == "lora_delta_w_svd":
+                continue
+
+            with self.subTest(combine_strategy=combine_strategy):
+                # 1. if we merge only one adapter, the weights of the new adapter should be the same as the original adapter
+                svd_rank = LoRAConfig().r if combine_strategy == "lora_delta_w_svd" else None
+                model.average_adapter(
+                    adapter_name=f"{combine_strategy}_case1",
+                    adapter_list=[f"{name}_0"],
+                    weights=[1],
+                    combine_strategy=combine_strategy,
+                    svd_rank=svd_rank,
+                )
+
+                filter_keys = [k.format(name=f"{combine_strategy}_case1") for k in ["loras.{name}."]]
+
+                if combine_strategy != "lora_delta_w_svd":
+                    for k, v in self.filter_parameters(model, filter_keys).items():
+                        adapter_0_key = k.replace(f"{combine_strategy}_case1", f"{name}_0")
+                        self.assertTrue(torch.allclose(v, adapter_0[adapter_0_key]))
+                else:
+                    # For lora_delta_w_svd, we need to calculate the expected weights since lora_delta_w_svd performs an SVD
+                    for i, layer in model.iter_layers():
+                        for module in layer.modules():
+                            if isinstance(module, LoRALayer):
+                                if f"{name}_0" in module.loras and f"{combine_strategy}_case1" in module.loras:
+                                    original_lora = module.loras[f"{name}_0"]
+                                    merged_lora = module.loras[f"{combine_strategy}_case1"]
+
+                                    # Compute SVD of the original delta_w
+                                    u, s, v = torch.svd(original_lora.delta_w)
+                                    u = u[:, :svd_rank]
+                                    s = s[:svd_rank]
+                                    v = v[:, :svd_rank]
+
+                                    # Reconstruct A and B matrices
+                                    expected_A = v.t()
+                                    expected_B = u @ torch.diag(s)
+
+                                    # Compare with merged adapter
+                                    self.assertTrue(torch.allclose(expected_A, merged_lora.lora_A, atol=1e-5))
+                                    self.assertTrue(torch.allclose(expected_B, merged_lora.lora_B, atol=1e-5))
+
```

Comment:
maybe move the svd part to a helper method since it is repeated multiple times in this test class?